src/EnergyPlus/HybridUnitaryAirConditioners.cc

```diff
@@ -0,0 +1,728 @@
+				)
+		{
+			// SUBROUTINE INFORMATION:
+			//       AUTHOR         Spencer Maxwell Dutton
+			//       DATE WRITTEN   October 2017  
+			//       MODIFIED       
+			//       RE-ENGINEERED  na
+
+			// PURPOSE OF THIS SUBROUTINE:
+			// 
+
+			// METHODOLOGY EMPLOYED:
+			// 
+
+			// REFERENCES:
+			// na
+
+			// Using/Aliasing
+			using InputProcessor::FindItemInList;
+			using General::TrimSigDigits;
+			// SUBROUTINE LOCAL VARIABLE DECLARATIONS:
+			int CompNum;
+			bool errorsfound = false;
+			if (GetInputZoneHybridEvap) {
+				GetInputZoneHybridUnitaryAirConditioners(errorsfound);
+				GetInputZoneHybridEvap = false;
+			}
+
+			if (CompIndex == 0) {
+				CompNum = FindItemInList(CompName, ZoneHybridUnitaryAirConditioner);
+				if (CompNum == 0) {
+					ShowFatalError("SimZoneHybridUnitaryAirConditioners: Zone evaporative cooler unit not found.");
+				}
+				CompIndex = CompNum;
+			}
+			else {
+				CompNum = CompIndex;
+				if (CompNum < 1 || CompNum > NumZoneHybridEvap) {
+					ShowFatalError("SimZoneHybridUnitaryAirConditioners: Invalid CompIndex passed=" + TrimSigDigits(CompNum) + ", Number of units =" + TrimSigDigits(NumZoneHybridEvap) + ", Entered Unit name = " + CompName);
+				}
+				if (CheckZoneHybridEvapName(CompNum)) {
+					if (CompName != ZoneHybridUnitaryAirConditioner(CompNum).Name) {
+						ShowFatalError("SimZoneHybridUnitaryAirConditioners: Invalid CompIndex passed=" + TrimSigDigits(CompNum) + ", Unit name=" + CompName + ", stored unit name for that index=" + ZoneHybridUnitaryAirConditioner(CompNum).Name);
+					}
+					CheckZoneHybridEvapName(CompNum) = false;
+				}
+			}
+			try
+			{
+				InitZoneHybridUnitaryAirConditioners(CompNum, ZoneNum);
+			}
+			catch (int e)
+			{
+				ShowFatalError("An exception occurred in InitZoneHybridUnitaryAirConditioners" + TrimSigDigits(CompNum) + ", Unit name=" + CompName + ", stored unit name for that index=" + ZoneHybridUnitaryAirConditioner(CompNum).Name+ ". Please check idf.");
+				return;
+			}
+			try
+			{
+				CalcZoneHybridUnitaryAirConditioners(CompNum, ZoneNum, SensibleOutputProvided, LatentOutputProvided);
+			}
+			catch (int e)
+			{
+				ShowFatalError("An exception occurred in CalcZoneHybridUnitaryAirConditioners" + TrimSigDigits(CompNum) + ", Unit name=" + CompName + ", stored unit name for that index=" + ZoneHybridUnitaryAirConditioner(CompNum).Name + ". Please check idf.");
+				return;
+			}
+			try
+			{
+				ReportZoneHybridUnitaryAirConditioners(CompNum);
+			}
+			catch (int e)
+			{
+				ShowFatalError("An exception occurred in ReportZoneHybridUnitaryAirConditioners" + TrimSigDigits(CompNum) + ", Unit name=" + CompName + ", stored unit name for that index=" + ZoneHybridUnitaryAirConditioner(CompNum).Name + ". Please check idf.");
+				return;
+			}
+		}
+		// this is used excusively for unit testing to get access to the ZoneHybridUnitaryAirConditioner
+		Model* HandleToHybridUnitaryAirConditioner(int UnitNum)

```

Comment:
This function is just hanging in the global state.  I don't see what you gain by having this pointer function.  Why not just get the `ZoneHybridUnitaryAirConditioner(UnitNum)` variable directly in your unit test??  I think this needs to go.

Reply:
ok done